### PR TITLE
Delay setting children style properties

### LINKED ISSUE #75: Unable to get rid of width when moving component out of SplitLayout
This is an issue in Vaadin 14.1.17

I'm using AppLayout, and on the right-hand side I'm going to have a silimar drawer to that AppLayout has on the left.

I have a button to toggle it on/off.
This takes the AppLayout content and replaces with new SplitLayout(appLayout.getContent(), rightDrawer);
When hiding, I do appLayout.setContent(splitLayout.getPrimaryComponent())
This works fine, *but* as soon as I set the split position, I can't get rid of it again.

When I create the split, the left side get "width: 80%" and the right get "width: 20%", which is expected.
When I remove the split, and set left side of split as appLayout content, the "width: 80%" remains.
It doesn't help to do setWidthFull()

If I set border as part of the toggle, that *is* propagated to the client, and if I use debugger check the content of getElement().getStyle() it has the expected width=100%.
It seems it is reset to 80% again after my split routine is done, but by what?

[SplitView.zip](https://github.com/vaadin/platform/files/4262487/SplitView.zip)

### PR DESCRIPTION
**Pull request by tulioag**
Fixes #75

### CODE REVIEW

File: vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java [hunk 1]
```diff
@@ -290,24 +318,18 @@ public void setSplitterPosition(double position) {
...
-                    + "if(this.children[i].slot === 'primary'"
-                    + "   || this.children[i].slot === 'secondary')"
-                    + "this.children[i].style.flex = ''";
-                getElement().executeJs(JS);
-                this.clearStylesRegistration = null;
-            });
+        return ui.beforeClientResponse(this, execution);
     }
 
+
```
Comment (hongduc-phan) [inline vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java:332]: Nit: shout we have the new empty line?

Comment (tulioag) [reply vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java]: Thanks. Fixed.

File: vaadin-split-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/splitlayout/test/LayoutComponent.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.HasSize;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.splitlayout.SplitLayout;
+
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
```
[the remark this answers is not in the document]
Comment (tulioag) [reply vaadin-split-layout-flow-integration-tests/src/main/java/com/vaadin/flow/component/splitlayout/test/LayoutComponent.java:2]: Thanks. Fixed.

File: vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java [hunk 3]
```diff
@@ -280,7 +283,32 @@ public Component getSecondaryComponent() {
      * @param position the relative position of the splitter, in percentages
      */
     public void setSplitterPosition(double position) {
-        double primary = Math.min(Math.max(position, 0), 100);
+        this.splitterPosition = position;
+        getUI().ifPresent(this::requestStylesUpdatesForSplitterPosition);
+    }
+
+    private void requestStylesUpdatesForSplitterPosition(UI ui) {
+        this.updateStylesRegistration = debounce(this.updateStylesRegistration,
```
Comment (alvarezguille) [inline vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java:291]: does it make sense to pass the registration by reference here? 
maybe it's late, but I think something is looks more complicated than it should
we can check next week

Comment (tulioag) [reply vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java]: Removed method.

**Pull request merged**